lib/core/include/dstream.hpp

```diff
@@ -435,6 +435,15 @@ namespace irods::experimental::io
     template <typename T>
     inline static constexpr std::ios_base::openmode default_openmode<std::basic_iostream<T>> = std::ios_base::in | std::ios_base::out;
 
+    template <typename>
```

Comment:
I remember us discussing this particular change... did we determine that this is more correct, or should we leave an TODO comment which explains that this may or may not move back inside the `basic_dstream` class?

Reply:
I don't think we made a decision one way or the other. Personally, I'm fine with leaving it outside the class, but I don't have a strong opinion on the matter.

Reply:
Thoughts, @korydraughn?

Reply:
I don't like them being public given that they are an implementation detail, but it is fine for now. We can always come back and research this later.

@SwooshyCueb What was the reason for this again? It was because of a GCC compilation issue, right?

Reply:
Never mind, I just read your note about the GCC bug.

@alanking So, we can restore the location of the templates once GCC resolves the issue.

Reply:
So let me make sure we're all on the same page here.  

Considering one of the goals is to not have to carry around our own compiler, if the decision is to make this scope/visibility change temporary until the parsing bug in GCC is fixed, we're going to be waiting a while before we can revert this change.

This workaround will stick around until:
1. GCC has implemented a fix for their parsing bug
2. The fix for this parsing bug has made it to distribution package repositories
3. We have dropped support for distributions that do not have a GCC with the fix in their repositories

All this taken into account...  
Should we proceed with the intention of reverting this fix however many years in the future?  
Should we consider making this workaround more permanent?  
Should we perhaps instead investigate a different solution?

Reply:
Per our previous discussion, I've moved `mandatory_openmode` into a detail namespace, so I think we can mark this as resolved